freeipa/src/main/java/com/sequenceiq/freeipa/client/FreeIpaClient.java

```diff
@@ -187,6 +188,16 @@ public void addPasswordExpirationPermission(String permission) throws FreeIpaCli
         invoke("permission_add", flags, params, Object.class);
     }
 
+    public int getUsernameLength() throws FreeIpaClientException {
+        Config config = (Config) invoke("config_show", List.of(), Map.of(), Config.class).getResult();
+        return config.getIpamaxusernamelength();
```

Comment:
May throw NPE. Possibly the return type of `getUsernameLength()` was supposed to be an `Integer`.

Reply:
If the config fetch fails then it will throw an exception, and this parameter is mandatory in FreeIPA, so if we have the Config object, then it must be present